### PR TITLE
Change reference of SeedGen and Serialization

### PR DESCRIPTION
**Pull request by sampathweb**

### CODE REVIEW

File: keras_cv/backend/random.py [hunk 1]
```diff
@@ -36,6 +38,13 @@ def next(self, ordered=True):
             self._current_seed[0] += 1
             return self._current_seed[:]
 
+    def get_config(self):
+        return {"seed": self._seed, **self._kwargs}
```
Comment (divyashreepathihalli) [inline keras_cv/backend/random.py:42]: you would not need `**self._kwargs`

Comment (sampathweb) [reply keras_cv/backend/random.py:42]: Why?  If the user passes kwargs we want to capture that in the config that's serializied.

Comment (divyashreepathihalli) [reply keras_cv/backend/random.py:42]: adding reference from keras 3 here - https://github.com/keras-team/keras/pull/18933

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]
LGTM! Hey Ramesh just one change requested.

File: keras_cv/backend/random.py (hunk 1 above)
Comment (sampathweb) [reply keras_cv/backend/random.py:42]: The reconstructed object will not be same as one originally created if we don't serialize the kwargs as well

File: keras_cv/layers/regularization/drop_path.py [hunk 2]
```diff
@@ -50,7 +50,11 @@ class DropPath(keras.layers.Layer):
     def __init__(self, rate=0.5, seed=None, **kwargs):
         super().__init__(**kwargs)
         self.rate = rate
-        self.seed = keras.random.SeedGenerator(seed=seed)
+        self._seed_val = seed
+        if not isinstance(seed, random.SeedGenerator):
```
Comment (divyashreepathihalli) [inline keras_cv/layers/regularization/drop_path.py:54]: The seed value that is expected for drop_path is an `integer`. This check is unnecessary, we should directly initialize the `SeedGenerator` with the seed.

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]

File: keras_cv/backend/random.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_cv/backend/random.py]: Resolving based on offline discussion. Adding `kwargs` to get_config is not recommended because we have no way to guarantee the provided args would be serializable or not.

File: keras_cv/layers/regularization/drop_path.py (hunk 2 above)
Comment (sampathweb) [reply keras_cv/layers/regularization/drop_path.py]: Done
